webplugin/js/app/mck-sidebox-1.0.js

```diff
@@ -9718,6 +9717,41 @@ var userOverride = {
                 }
                 return '';
             };
+            _this.getMessageCreatedAtTime = function (createdAtTime) {
+                if (TIME_FORMAT_24_HOURS) {
+                    var msgtime = new Date(createdAtTime);
+                    var yesterday = new Date();
+                    if (yesterday.getDate() != msgtime.getDate()) {
+                        var time = msgtime.toLocaleString('en-US', {
+                            day: 'numeric',
+                            month: 'short'
+                        }) + ", " + msgtime.getHours() + ":" + msgtime.getMinutes();
+                        return time;
+                    }
+                    else if (yesterday.getDate() != msgtime.getDate() && yesterday.getMonth() != msgtime.getMonth()) {
```

Comment:
something looks wrong in this code
if yesterday.getDate() != msgtime.getDate() is true then it will go inside the 'if' condition above it, and will never even come to this else if statement.

Reply:
committed new code

so this is what I have thought
if date does not match - then go inside if and show date and time
if date matches but month does not mach - then go inside else if and show date and time
if date matches and month also matches but year does not match - then go inside second else if and show date and time
else
show time only

Reply:
review it, conditions are not looking correct

Reply:
I changed the condition now - and committed new changes. still does not look good?